delete scenario which delete product package from package state

mcalmer opened the pull request:
## What does this PR change?

This scenario should not be needed since round about a year now.
Adding the products to the package state was dropped in https://github.com/uyuni-project/uyuni/pull/2158

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/pull/14287

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

srbarrios commented:
@mcalmer could we please don't merge not urgent PRs during the weekend?

If you would wait a bit, I would tell you that we are also removing this package in our automated MU and milestones validations, I would ask you to explain me with more detail the reason why we don't need to remove this package, and I would ask you a doubt I have with sles12sp4, which has two packages with this name, one of them finishing in -POOL.

mcalmer commented:
> @mcalmer thanks for taking care of CI, but please, try to wait a bit more for non urgent PRs, it's hard to keep track of PRs created and merged during the weekend.

A not full test run over the weekend is imporant for me. Sorry.
 
> If I would saw it before, I would tell you that we are also removing this package in our automated MU and milestones validations, and I would ask you to keep both testsuites synch on that kind of changes, I would ask you to explain me with more detail the reason why we don't need to remove this package, and I would ask you a doubt I have with sles12sp4, which has two packages with this name, one of them finishing in -POOL.

As we can now see, there is also another test we need to check.

  `Remove sle-manager-tools-release from state after bootstrap via XML-RPC`

This is the reason why I added you all as reviewer. We can get all this removed at all places where it happens.
I will try to come up with another PR.

About the reason why this is not required anymore.

In the past (over a year ago) we detected the products during bootstrapping and added the release packages to the Package State to get them installed. This did not work so well as it required that the channels were added before this and this could not be guaranteed as it was an async task.

We implemented a different mechnism and wrote a salt module which is always part of the channel state and add the required product packages automatically. At that time the Package State was not required anymore. Some weeks later, we removed the code from bootstrapping which generate the package state.

This means you can wait as long as you want, you will never see `suse-manager-tools-release` package in the Package State.
I am not sure why the state was working before. Maybe we have something like: "it is ok when the package is not found" ?

Important: the package is still installed, but not via a Package State. When there are tests which remove the package, they might still be required. But cleaning up the Package State is not required anymore.
